src/main/java/seedu/address/storage/JsonAdaptedItem.java

```diff
@@ -93,7 +98,16 @@ public Item toModelType() throws IllegalValueException {
         final Supplier modelSupplier = new Supplier(supplier);
 
         final Set<Tag> modelTags = new HashSet<>(itemTags);
-        return new Item(modelName, modelQuantity, modelSupplier, modelTags);
+
+        if (maxQuantity == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Quantity.class.getSimpleName()));
+        }
```

Comment:
Perhaps not checking is fine too? Since we do not require max quantity to be a mandatory field. Having this check will go against our design